services/QuillLMS/spec/controllers/teachers/unit_activities_controller_spec.rb

```diff
@@ -24,6 +24,11 @@
       put :hide, params: { id: unit_activity.id }
       expect(unit_activity.reload.visible).to eq false
     end
+
+    it 'should touch the parent unit in order to bubble up cache clearing' do
+      expect_any_instance_of(Unit).to receive(:touch)
```

Comment:
In general I try to avoid using `expect_any_instance` and `allow_any_instance` as they are legacy methods that the rspec maintainers [advocate](https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance) not using.

For this one you could stub the class finder method that gets you the object:
```
      allow(UnitActivity).to receive(:find).with(unit_activity.id.to_s).and_return(unit_activity)
      expect(unit_activity.unit).to receive(:touch)
```
The benefit of this approach is that we're specifically targeting one instance rather than all instances.

There is a [cop](https://www.rubydoc.info/gems/rubocop-rspec/1.7.0/RuboCop/Cop/RSpec/AnyInstance) for this one, but I think we currently have too many violations so I haven't turned it on yet.